osbuild/api.py

```diff
@@ -176,9 +177,14 @@ def _setup_stdio(self, server):
 
             server.send(msg, fds=fds)
 
+    def _set_metadata(self, message):
+        self.metadata.update(message["metadata"])
+
     def _message(self, msg, fds, sock):
         if msg["method"] == 'setup-stdio':
             self._setup_stdio(sock)
+        elif msg["method"] == 'set-metadata':
```

Comment:
Maybe `add-metadata`? To me `set` implies that you override previous data. But... not very important.

Reply:
Ah, yeah, I think that this is a better name indeed.